frontend/packages/dev-console/src/components/health-checks/AddHealthCheck.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { FirehoseResult, ContainerDropdown, history } from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FormFooter } from '@console/shared';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import HealthChecks from './HealthChecks';
+
+type AddHelathProps = {
+  resource?: FirehoseResult<K8sResourceKind>;
+  currentContainer: string;
+};
+
+const AddHealthCheck: React.FC<FormikProps<FormikValues> & AddHelathProps> = ({
+  resource,
+  currentContainer,
+  handleSubmit,
+  handleReset,
+  dirty,
+  errors,
+  status,
+  isSubmitting,
+  setFieldValue,
+}) => {
+  const { data } = resource;
+  const [currentKey, setCurrentKey] = React.useState(currentContainer);
+
+  const handleSelectContainer = (containerName) => {
+    setCurrentKey(containerName);
+    setFieldValue('containerName', containerName);
+    history.push(
+      `/k8s/ns/${data.metadata.namespace}/${data.kind || data.kind}/${
+        data.metadata.name
+      }/containers/${containerName}/health-checks`,
+    );
+  };
+
+  const containers = resource.loaded && _.keyBy(data.spec.template.spec.containers, 'name');
+
+  return (
+    <div className="co-m-pane__body">
+      <Form onSubmit={handleSubmit}>
+        <div>
+          Managing health check for:{'  '}
+          <ContainerDropdown
+            currentKey={currentKey}
+            containers={containers}
+            onChange={handleSelectContainer}
+          />
+        </div>
+        <HealthChecks resourceType={getResourcesType(data)} />
+        <FormFooter
+          handleReset={handleReset}
+          errorMessage={status && `${status.errors}`}
```

Comment:
Why convert status.error to string?

Reply:
Because of `errorMessage` except string and I am getting the error in json format.

Reply:
@vikram-raj if error isn't a string how can it be displayed properly?